libraries/ESP8266WiFi/src/WiFiClient.cpp

```diff
@@ -137,6 +137,22 @@ int WiFiClient::connect(const String& host, uint16_t port)
     return connect(host.c_str(), port);
 }
 
+int WiFiClient::connectAsync(const char* host, uint16_t port, uint8_t* waiting) // Non blocking
```

Comment:
Should be a `bool` and return `false` or `true`

Reply:
The function has the same formal parameters and return type as WiFiClient::connect.
I just added the waiting parameter. Since they are similar I've not changed it for consistency.